### PR TITLE
Remove CUIICatalog invalid asset name supplied error

### PR DESCRIPTION
**Pull request by mmckinley8**
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request:

(You can skip this if you're fixing a typo or adding an app to the Showcase.)

Explain the **motivation** for making this change. What existing problem does the pull request solve?

Prefer **small pull requests**. These are much easier to review and more likely to get merged. Make sure the PR does only one thing, otherwise please split it.

**Test plan (required)**

Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

Make sure tests pass on both Travis and Circle CI.

**Code formatting**

Look around. Match the style of the rest of the codebase. See also the simple [style guide](https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md#style-guide).

For more info, see the ["Pull Requests" section of our "Contributing" guidelines](https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md#pull-requests).

**TabBarIOS icons were showing up as invalid strings, causing crashes on some devices.**

**Test plan**

Set a symbolic breakpoint in xcode -
Sybmol: [UIImage imageNamed:]
Condition: $arg3 == nil

If the app crashes with this breakpoint added, this PR will resolve it. You will no longer see:
CUIICatalog invalid asset name supplied (null)

### CODE REVIEW

**Comment by facebook-github-bot-0** [comment]
By analyzing the blame information on this pull request, we identified @nicklockwood and @ryangomba to be potential reviewers.

**Comment by facebook-github-bot-0** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by facebook-github-bot-2** [comment]
@mmckinley8 updated the pull request.

**Comment by facebook-github-bot-4** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

**Comment by ide** [comment]
Add an explicit nil check to fix the root cause instead of patching the symptom.

File: React/Base/RCTConvert.m [hunk 1]
```diff
@@ -882,7 +882,7 @@ + (UIImage *)UIImage:(id)json
   NSString *scheme = URL.scheme.lowercaseString;
   if ([scheme isEqualToString:@"file"]) {
     NSString *assetName = RCTBundlePathForURL(URL);
-    image = [UIImage imageNamed:assetName];
+    image = [UIImage imageNamed:[NSString stringWithFormat:@"%@", assetName]];
```
Comment (nicklockwood) [inline React/Base/RCTConvert.m]: I agree with @ide. I suggest changing this to:

    image = assetName ? [UIImage imageNamed:assetName] : nil;

Comment (mmckinley8) [inline React/Base/RCTConvert.m]: Totally agree, thanks @ide and @nicklockwood ...first time submitting PR to this repo, why did I not pass the The Travis CI?

**Comment by facebook-github-bot-1** [comment]
@mmckinley8 updated the pull request.

**Comment by facebook-github-bot-4** [comment]
@mmckinley8 updated the pull request.

**Comment by facebook-github-bot-5** [comment]
@mmckinley8 updated the pull request.

**Comment by facebook-github-bot-5** [comment]
@mmckinley8 updated the pull request.

**Comment by facebook-github-bot-4** [comment]
It's been a while since the last commit was reviewed and the labels show this pull request needs review. Based on the blame information for the files in this pull request we identified @nicklockwood as a potential reviewer. Could you take a look please or cc someone with more context?

**Comment by ide** [comment]
@facebook-github-bot shipit

**Comment by atkit** [comment]
Still happens in `RCTImageFromLocalAssetURL`

**Pull request closed without merging**